internal/bminventory/inventory.go

```diff
@@ -518,6 +518,18 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 			"Openshift version %s support level is: %s, and can't be used for creating a new cluster",
 			swag.StringValue(params.NewClusterParams.OpenshiftVersion), releaseImage.SupportLevel))
 	}
+	// Ensure a relevant OsImage exists. For multiarch we disabling the code below because we don't know yet
+	// what is going to be the architecture of InfraEnv and Agent.
+	if len(releaseImage.CPUArchitectures) == 1 {
+		releaseVersion := *releaseImage.OpenshiftVersion
+		releaseArch := releaseImage.CPUArchitectures[0]
+		var osImage *models.OsImage
+		osImage, err = b.osImages.GetOsImage(releaseVersion, releaseArch)
+		if err != nil || osImage.URL == nil {
```

Comment:
URL property existence should actually be ensured on osimages [validate](https://github.com/openshift/assisted-service/blob/3c7a420f984202790ddbbc18e2b8aa04bbc392e4/internal/versions/osimages.go#L54). Did we encounter a null osImage.URL in some flow?